test/sidekick_test.exs

```diff
@@ -2,6 +2,18 @@ defmodule SidekickTest do
   use ExUnit.Case, async: false
 
   test "starts a remote node" do
-    assert Sidekick.start([]) == :ok
+    assert Sidekick.start(:test, []) == :ok
+
+    # Sidekick node should exist
+    assert Node.list() == [:"test@127.0.0.1"]
+
+    # Supervisor should be started on the Node
+    pid = :rpc.call(:"test@127.0.0.1", Process, :whereis, [Sidekick.Supervisor])
```

Comment:
I think that `GenServer.whereis({Sidekick.Supervisor, :"test@127.0.0.1"})` would also work.